### ISSUE TITLE
UI/UX Improvement of Task Details Page

### ISSUE THREAD

**Issue by ardourApeX** [Real-Dev-Squad/website-status#996]
### Description:
Currently, the UI of the task details page on the status site does not seem to be satisfactory. It lacks in terms of UI, UX, features accessibility.

### Requirements:
To enhance the UI and UX of the task details page, the following modifications/improvements should be implemented:

**Comment by vikhyat187** [comment]
Hey @ardourApeX the first one, is that done by fuzzy search?

**Comment by prakashchoudhary07** [comment]
@ardourApeX, For search in the input box, by default, it should be task search, and if the assignee is typed then the users search takes place, based on the applied filter the suggestion is given, or else its default task
Can we do this?

**Comment by ardourApeX** [comment]
> Hey @ardourApeX the first one, is that done by fuzzy search?

- The first one can be implemented using hard-coded logic. For example, if a user is typing, then we can show all the possible keys that have not been selected by the user yet. 
- For the second one, fuzzy search can be implemented. Although Firebase does not have any built-in features for that, we may have to use third-party services.

**Comment by ardourApeX** [comment]
> @ardourApeX, For search in the input box, by default, it should be task search, and if the assignee is typed then the users search takes place, based on the applied filter the suggestion is given, or else its default task We also show the suggestion for the filter keys also? Can we do this?


I'm not sure if I understood your question correctly, but based on what I understand, here is my answer:

The purpose of showing suggestions here is to minimize errors. The more we ask the user to type, the greater the chances of errors. To overcome this issue, I thought of auto-completing the key part based on the typed value. It will involve hard-coded logic since we have a limited number of keys.

As per my understanding, we use the key for status and assignee, but not for the title. However, to make it more consistent and understandable in the UI, I thought of adding a title key as well. 

**One thing we can do for the UI purpose is to have a title key and keep the API calling structure the same as before.** So that the functionality will remain the same for each use case.

**Comment by bhtibrewal** [comment]
will this be scalable if we have more fields in the future, 
also can you state the technical aspects of how we are going to achieve these functionalities

**Comment by ardourApeX** [comment]
> can you share, how will this be scaled if we have more fields in the future, also can you state the technical aspects of how we are going to achieve these functionalities

[Here is the POC](https://codesandbox.io/s/clever-breeze-lfwf29?) I did. It is not end-to-end working but may depict the actual implementation.
